src/core/keychain/keychainTypes/hdKeychain.ts

```diff
@@ -87,9 +99,10 @@ export class HdKeychain implements IKeychain {
 
   getSigner(address: Address): Signer {
     const _privates = privates.get(this)!;
-    const wallet = _privates.getWalletForAddress(address);
-    if (!wallet?._isSigner) throw new Error('Not a signer');
-    return wallet;
+    const provider = getProvider({ chainId: mainnet.id });
```

Comment:
i checked l2 sends because of this, working fine, but chainId as mainnet doesn't do anything?

Reply:
The interface of signer requires to pass a provider to the constructor. That's why it's set to mainnet but when used through Wallet we always pass the right provider